main/webservices/api/test_api.php

```diff
@@ -2,116 +2,550 @@
+     */
+    protected function integer($action, $parameters = [])
+    {
+        $value = $this->singleElementValue($action, $parameters);
+        $this->assertIsInt($value);
+        return $value;
+    }
+
+    /**
+     * Posts an action request and assert it returns an array with a single boolean value
+     *
+     * @param string    $action         name of the webservice action
+     * @param array     $parameters     parameters to send with the request
+     * @return boolean                  the boolean value
+     */
+    protected function boolean($action, $parameters = [])
+    {
+        $value = $this->singleElementValue($action, $parameters);
+        $this->assertIsBool($value);
+        return $value;
+    }
+
+    /**
+     * @throws Exception if it cannot delete the created test user
+     */
+    public function testCreateUser()
+    {
+        $ACTION = 'save_user';
+
+        // create a user
+        $EXTRA_FIELD_NAME = 'age';
+        $EXTRA_FIELD_VALUE = '29';
+        $ORIGINAL_VALUES = [
+            'firstname' => 'Test User'.time(),
+            'lastname' => 'Chamilo',
+            'status' => 5, // student
+            'email' => 'testuser@example.com',
+            'loginname' => 'restuser'.time(),
+            'password' => 'restuser',
+            'original_user_id_name' => 'myplatform_user_id', // field to identify the user in the external system
+            'original_user_id_value' => '1234', // ID for the user in the external system
+            'language' => 'english',
+            //'phone' => '',
+            //'expiration_date' => '',
+        ];
+        $params = [];
+        foreach($ORIGINAL_VALUES as $name => $value) {
+            $params[strtolower($name)] = $value;
+        }
+        $params['extra'] =  [
+            [ 'field_name' => $EXTRA_FIELD_NAME, 'field_value' => $EXTRA_FIELD_VALUE ],
+        ];
+        $userId = $this->integer($ACTION, $params);
+
+        // compare each saved value to the original
+        $user = UserManager::getManager()->find($userId);
+        foreach($ORIGINAL_VALUES as $name => $value) {
+            $methodName = 'get'.ucfirst($name);
+            if ($name !== 'password' and method_exists($user, "$methodName")) {
+                $this->assertSame($value, eval("return \$user->$methodName();"), $name);
+            }
+        }
+
+        // assert user extra field value was saved
+        $extraFieldValueModel = new ExtraFieldValue('user');
+        $extraFieldValue = $extraFieldValueModel->get_values_by_handler_and_field_variable($userId, $EXTRA_FIELD_NAME);
+        $this->assertNotFalse($extraFieldValue);
+        $this->assertSame($EXTRA_FIELD_VALUE, $extraFieldValue['value']);
+
+        // clean up
+        UserManager::delete_user($userId);
+    }
+
+    /**
+     * @throws Exception if it cannot add the test courses to the test model session
+     */
+    public function testCreateSessionFromModel()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The variable $COURSE_CODES is not named in camelCase.](https://app.codacy.com/app/chamilo/chamilo-lms/pullRequest?prid=4554450)